### DISCUSSION TITLE
How to ensure high availability of apisix when apisix as a the Front gateway

### DISCUSSION THREAD

**Discussion by carverLiu** [apache/apisix#6060] [General]
i am very pleasureto meet apisix ，and ask some question for  high availability of apisix
1、First ,apisix as a the Front gateway , when the apisix service down ,What should I do for it?
    i think may be i can use keepalived and Vip to listen the master ,when the master down , the slave can get the message by heartbeat and the slave become the matser 。
    but the idea drive other problem, the apisix can loss some traffic When the slave node listens to the downtime of the master node，may be 2~4s 。
SO.....I hope dear friends can give some other advise to slove this problem

**Comment by carverLiu** [comment]
中文：我们的业务需要使用apisix作为前置一级网关，当apisix 宕机了，我如何保证当前业务的可用性呢？
目前想到的是使用keealived+vip实现主从热备，故障自动转移，slave 成为master ，虚拟IP漂移地址到slave。
但是这种方法，故障自动恢复的过程需要2~4S恢复服务，导致这段时间流量丢失，各位大大有什么建议嘛？

**Reply by soulsoul** [reply]
知道 api7 么   商业版有集群高可用方案

**Reply by carverLiu** [reply]
> 求大大提供一个链接

**Reply by leslie-tsang** [reply]
Check [this](https://www.apiseven.com/zh/products/api7/features) out.
